Classes Attribute, AttributeValue, ProductItem

liqueflies opened the pull request:
Add new files

liqueflies committed:
Classes Attribute, AttributeValue, ProductItem

picce commented on projects/PigeonCms.Core/BLL/Attribute.cs:
```diff
@@ -0,0 +1,128 @@
+
+namespace PigeonCms
+{
+    class Attribute: ITableWithOrdering
```
use ITable interface

picce commented on projects/PigeonCms.Core/BLL/Attribute.cs:
```diff
@@ -0,0 +1,128 @@
+
+        #endregion
+
+        public class ItemComparer : IComparer<Item>
```
non serve ItemComparer

picce commented on projects/PigeonCms.Core/BLL/AttributeValue.cs:
```diff
@@ -0,0 +1,139 @@
+
+namespace PigeonCms
+{
+    class AttributeValue : ITableWithOrdering
```
use ITable

picce commented on projects/PigeonCms.Core/BLL/AttributeValue.cs:
```diff
@@ -0,0 +1,139 @@
+{
+    class AttributeValue : ITableWithOrdering
+    {
+        const string DefaultItemType = "PigeonCms.AttributeValue";
```
non serve

picce commented on projects/PigeonCms.Core/BLL/Attribute.cs:
```diff
@@ -0,0 +1,128 @@
+{
+    class Attribute: ITableWithOrdering
+    {
+        const string DefaultItemType = "PigeonCms.Attribute";
```
non serve default value, può essere vuoto

picce commented on projects/PigeonCms.Core/BLL/Attribute.cs:
```diff
@@ -0,0 +1,128 @@
+        /// Item specific type name. Ex. PigeonCms.CustomItem
+        /// </summary>
+        [DataObjectField(false)]
+        public string ItemTypeName
```
rename in ItemType

picce commented on projects/PigeonCms.Core/BLL/AttributeValue.cs:
```diff
@@ -0,0 +1,139 @@
+        const string DefaultItemType = "PigeonCms.AttributeValue";
+        private int id = 0;
+        private int attributeId = 0;
+        private string itemType = "";
```
non serve itemType

picce commented on projects/PigeonCms.Core/BLL/AttributeValue.cs:
```diff
@@ -0,0 +1,139 @@
+        /// Item specific type name. Ex. PigeonCms.CustomItem
+        /// </summary>
+        [DataObjectField(false)]
+        public string ItemTypeName
```
non serve

picce commented:
#__attributesValues.value --> valueString (json serialized)
#__itemsAttributesValues.customValue --> customValueString (json serialized)

In class BLL aggiungere prop GET Value e CustomValue con visualizzano il valore in current culture

liqueflies committed:
Correction classes and managers

Ho corretto i bug delle classi ed ho abbozzato il discorso della lingua,
devo rivedere nel momento in cui avrò tutti i managers. Ho fatto i
manager si attribute e attributeValue, in attribute nel delete cancella
tutti i suoi value, che a loro volta dovrebbero cancellare le righe
nella tabella che relaziona il tutto con gli item, solo che non ho
ancora fatto il manager di quella per un unico dubbio: la chiave
primaria è doppia, come si fa a settare? Controlla pure e fammi sapere!
